### PR TITLE
[Doc] Better backport/branch label documentation

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR improves the backport label documentation.

### Motivation

Better doc.

### CODE REVIEW

File: docs/dev/contributing.md [hunk 1]
```diff
@@ -293,6 +293,8 @@ labels that can be use:
 - `backport/<branch-name>`: Add this label to automatically create a PR against the `<branch-name>` branch with your backported changes. The backport PR creation is triggered:
   - When a PR with the label is merged
   - When an already-merged PR gets the label
+
+  In case of conflicts, the bot will prompt a list of instructions to follow ([example](https://github.com/DataDog/datadog-agent/pull/23316#issuecomment-1973207164)) to manually backport your PR.
```
Comment (maycmlee) [inline docs/dev/contributing.md:297]:
```suggestion
  If there is a conflict, the bot prompts you with a list of instructions to follow ([example](https://github.com/DataDog/datadog-agent/pull/23316#issuecomment-1973207164)) to manually backport your PR.
```

**Review by maycmlee — COMMENTED** [review]
Just a couple of suggestions, let me know if that is not the correct meaning.

**Comment by amenasria** [comment]
@maycmlee makes sense, commited it, thanks !

**Review by maycmlee — APPROVED** [review]

**Review by CelianR — APPROVED** [review]
Lgtm

**Pull request merged**